src/Controller/UserController.php

```diff
@@ -60,4 +60,29 @@ public function addFavoriteBook(#[CurrentUser] ?User $user, string $id): Respons
 
     return $this->json(['success' => true, 'data' => $book]);
   }
+
+  public function removeFavoriteBook(#[CurrentUser] ?User $user, string $id): Response 
+  {
+    if ($user === null) {
+      return $this->json(['message' => 'missing or wrong credentials'], Response::HTTP_UNAUTHORIZED);
+    }
+
+    $book = $this->bookDBService->searchBookById($id);
+
+    if ($book === null) {
+      return $this->json(['message' => 'Book not found'], Response::HTTP_NOT_FOUND);
+    }
+
+    if (!$user->getFavorites()->contains($book)) {
+      return $this->json(['message' => 'Book is not in favorites'], Response::HTTP_BAD_REQUEST);
+    }
+
+    $user->removeFavorite($book);
+
+    $this->entityManager->persist($user);
+    $this->entityManager->flush();
+
+    return $this->json(['success' => true, 'data' => $book]);
+
+  }
```

Comment:
I would rather return the new favoritesList instead of the book itself.
Like that our frontend will get the latest list and can update accordingly